xdsl/interactive/app.py

```diff
@@ -213,17 +220,66 @@ def watch_available_pass_list(
+        """
+        This function facilitates user input of pass concatenated_arg_val by navigating
+        to the AddArguments screen, and subsequently parses the returned string upon
+        screen dismissal and appends the pass to the pass_pipeline variable.
+        """
+
+        def add_pass_with_arguments_to_pass_pipeline(concatenated_arg_val: str) -> None:
+            """
+            Called when AddArguments Screen is dismissed. This function attempts to parse
+            the returned string, and if successful, adds it to the pass_pipeline variable.
+            In case of parsing failure, the AddArguments Screen is pushed, revealing the
+            Parse Error.
+            """
+            try:
+                new_pass_with_arguments = list(
+                    parse_pipeline(
+                        f"{selected_pass_value.name}{{{concatenated_arg_val}}}"
+                    )
+                )[0]
+                self.pass_pipeline = (
+                    *self.pass_pipeline,
+                    (selected_pass_value, new_pass_with_arguments),
+                )
+
+            except PassPipelineParseError as e:
+                res = f"PassPipelineParseError: {e}"
+                screen = AddArguments(TextArea(res, id="argument_text_area"))
+                self.push_screen(screen, add_pass_with_arguments_to_pass_pipeline)
+
+        # generates a string containing the concatenated_arg_val and types of the selected pass and initializes the AddArguments Screen to contain the string
+        self.push_screen(
+            AddArguments(
+                TextArea(
+                    get_pass_argument_names_and_types(selected_pass_value),
+                    id="argument_text_area",
+                )
+            ),
+            add_pass_with_arguments_to_pass_pipeline,
+        )
+
     @on(ListView.Selected)
     def update_pass_pipeline(self, event: ListView.Selected) -> None:
         """
         When a new selection is made, the reactive variable storing the list of selected
         passes is updated.
         """
         selected_pass = event.item.name
-        for name, value in ALL_PASSES:
-            if name == selected_pass:
-                self.pass_pipeline = tuple((*self.pass_pipeline, value))
-                return
+        for name_pass, value_pass in ALL_PASSES:
+            if name_pass == selected_pass:
+                # check if pass has concatenated_arg_val
+                if len(fields(value_pass)) != 0:
+                    # call function that handles adding passes with concatenated_arg_val to the pass_pipeline
+                    self.get_pass_arguments(value_pass)
+                else:
+                    # add the selected pass to pass_pipeline
+                    self.pass_pipeline = (
+                        *self.pass_pipeline,
+                        (value_pass, value_pass().pipeline_pass_spec()),
+                    )
```

Comment:
I would recommend folding this logic into `get_pass_arguments`, it feels like that function should be deciding whether to show a screen or not